src/Cardano/Shell/NodeIPC/ServerExample.hs

```diff
@@ -118,11 +124,14 @@ receieveMessages serverReadHandle = do
     reply   <- readServerMessage -- Reply
     return (started, reply)
 
-getHandleFromEnv :: String -> IO Handle
+getHandleFromEnv :: String -> IO (Either NodeIPCException Handle)
 getHandleFromEnv envName = do
     mFdstring <- lookupEnv envName
     case mFdstring of
-        Nothing -> throwM $ NodeChannelNotFound (strConv Lenient envName)
+        Nothing -> throw $ NodeChannelNotFound (strConv Lenient envName)
         Just fdstring -> case readEither fdstring of
-            Left err -> throwM $ UnableToParseNodeChannel (strConv Lenient err)
-            Right fd -> liftIO $ fdToHandle fd
+            Left err -> throw $ UnableToParseNodeChannel (strConv Lenient err)
+            Right fd -> Right <$> fdToHandle fd
+  where
+    throw :: NodeIPCException -> IO (Either NodeIPCException Handle)
```

Comment:
I didn't read that in context which is why I made the comment I made. I think it should be called just about anything other than `throw`, because that implies exceptions.